framework/src/engine/endpoint/txpool.ts

```diff
@@ -94,4 +108,44 @@ export class TxpoolEndpoint {
 	public async getTransactionsFromPool(_context: RequestContext): Promise<string[]> {
 		return this._pool.getAll().map(tx => tx.getBytes().toString('hex'));
 	}
+
+	public async dryRunTransaction(ctx: RequestContext): Promise<DryRunTransactionResponse> {
+		const reqErrors = validator.validate(dryRunTransactionRequestSchema, ctx.params);
+		if (reqErrors?.length) {
+			throw new LiskValidationError(reqErrors);
+		}
+		const req = (ctx.params as unknown) as DryRunTransactionRequest;
+		const transaction = Transaction.fromBytes(Buffer.from(req.transaction, 'hex'));
+
+		const { result } = await this._abi.verifyTransaction({
+			contextID: Buffer.alloc(0),
+			transaction: transaction.toObject(),
+		});
+		if (result === TransactionVerifyResult.INVALID) {
+			return {
+				success: false,
+				events: [],
+			};
+		}
+
+		const stateStore = new StateStore(this._blockchainDB);
+		const consensus = await this._consensus.getConsensusParams(
+			stateStore,
+			this._chain.lastBlock.header,
+		);
+
+		const response = await this._abi.executeTransaction({
+			contextID: Buffer.alloc(1),
```

Comment:
```suggestion
			contextID: Buffer.alloc(0),
```
dryRun doesn't require context, so it can use just 0 bytes here as well